core/src/sign.c

```diff
@@ -480,30 +481,39 @@ compute_output_hash(char xpub[static MULTISIG_PARTS][XPUB_SIZE],
  *
  * The code creates two signatures for a 2-of-4 segwit wallet. The result can't
  * be verified with Electrum but we can broadcast it on the Testnet.
+ *
+ * Confidential items have been marked.
+ * An effort has been made to zeroize them regardless of success or failure of this function.
+ *
  */
 Result handle_sign_tx(InternalCommandRequest_SignTxRequest *request,
                       InternalCommandResponse_SignTxResponse *response) {
+  Result r = Result_SUCCESS;
+  uint8_t CONFIDENTIAL seed[SHA512_DIGEST_LENGTH] = {0};
+  HDNode CONFIDENTIAL wallet;
+  memzero((void *) &wallet, sizeof(HDNode));
+  char CONFIDENTIAL xpub[MULTISIG_PARTS][XPUB_SIZE] = {0};
+  uint8_t CONFIDENTIAL public_key[33] = {0};
```

Comment:
any special reason for moving the declaration of `public_key` up here? the nice thing of declaring it later is that the scope is smaller (making it harder to introduce bugs).

Reply:
actually I moved everything up there to facilitate the do while pattern. I think we can move it back down there.

Reply:
Actually that public key is inside two for loops and the cleanup is outside all of them.  I would just keep it at the top.